coherence-pr-review.py

```diff
@@ -0,0 +1,266 @@
+
+
+# %% [markdown]
+# Testing the specifity_ne_spacy function :
+
+# %%
+specificity_ne_spacy(data.iloc[0]['Story'])
+
+# %%
+specificity_ne_spacy(data.iloc[1]['Story'])
+
+# %% [markdown]
+# # 2.Personal Context 
+
+# %%
+import nltk
+from nltk.sentiment import SentimentIntensityAnalyzer
+from gensim import corpora, models
+
+def evaluate_personal_context(text):
+    # Tokenize text into sentences
+    sentences = nltk.sent_tokenize(text)
+
+    # Sentiment Analysis
+    sia = SentimentIntensityAnalyzer()
+    sentiment_scores = [sia.polarity_scores(sentence)["compound"] for sentence in sentences]
+
+    # Topic Modeling
+    tokenized_sentences = [nltk.word_tokenize(sentence.lower()) for sentence in sentences]
+    dictionary = corpora.Dictionary(tokenized_sentences)
+    corpus = [dictionary.doc2bow(tokens) for tokens in tokenized_sentences]
+    lda_model = models.LdaModel(corpus, num_topics=3, id2word=dictionary)
+
+    # Extract most significant topics
+    topics = [lda_model.get_document_topics(doc) for doc in corpus]
+    most_significant_topics = [max(topic, key=lambda x: x[1]) for topic in topics]
+
+    # Get actual topics
+    actual_topics = [lda_model.print_topic(topic[0]) for topic in most_significant_topics]
+
+    # Return sentiment scores and actual topics
+    return sentiment_scores, actual_topics
+
+
+# %% [markdown]
+# In the context of topic modeling with LDA, the weights assigned to each word in a topic represent the importance or prevalence of that word within the topic. In the output you provided
+# 
+# Understanding the significance of individual words and their weights within a topic can help provide insights into the key themes and subjects present in the text.
+
+# %%
+def test_personal_context(text):
+    sentiment_scores, actual_topics = evaluate_personal_context(text)
+    print("Sentiment Scores:", sentiment_scores)
+    print()
+    print("Actual Topics:")
+    for topic in actual_topics:
+        print(topic)
+
+# %%
+test_personal_context(data.iloc[0]['Story'])
+
+# %%
+test_personal_context("i lost my best friend and im so happy")
+
+# %% [markdown]
+# the word "my" has the highest weight of 0.102 for the extracted topic.
+# 
+# A high weight for the word "my" suggests that it is a significant term within the topic identified by the LDA model. This means that the word "my" occurs frequently and carries substantial importance within the text when discussing the particular topic associated with that topic index.
+# 
+# In this case, it indicates that personal ownership or possession, likely related to the topic of loss and enduring emotional impact, plays a prominent role in the text. The word "my" may be indicating a personal connection or the speaker's individual perspective in relation to the topic being discussed.
+
+# %% [markdown]
+# ## 3.Sensory details 
+
+# %%
+import nltk
+
+def analyze_sensory_details(text):
+    sentences = nltk.sent_tokenize(text)
+
+    sensory_keywords = {
```

Comment:
This looks like a global declaration of sensory keywords.

Move this declaration to the const on the top level of a script. Or create a top-level variable and use it as an argument of a function. Another (better) option: move it to some config file in the proper format to support easy modifications.